llvm/test/CodeGen/SPIRV/hlsl-intrinsics/dot4add_i8packed.ll

```diff
@@ -0,0 +1,48 @@
+; RUN: llc -O0 -mtriple=spirv32v1.3-vulkan-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv32v1.3-vulkan-unknown %s -o - -filetype=obj | spirv-val %}
```

Comment:
You should be using `-mtriple=spirv-vulkan-unknown`. Also add tests that use Vulkan 1.3. `-mtriple=spirv-vulkan1.3-unknown` I believe. Also a test that enables the extension: `--spirv-ext=+SPV_KHR_integer_dot_product`.

You might need changes to SPIRVModuleAnalysis.cpp to get this to work.

Reply:
Updated the testcases to test the different cases. I seem required to use `spirv[32|64]` over `spirv` to meet the `Int8` capability requirements.

Reply:
You should be using `spirv`. You probably need to update `initAvailableCapabilitiesForVulkan` to allow the `Int8` capability. That is core in every version of spir-v, so you can make it always available.